### DISCUSSION TITLE
How well does Dependency-Track scale?

### DISCUSSION THREAD

**Discussion by AbdelHajou** [DependencyTrack/dependency-track#1540] [Q&A] [answered]
Hi all,

I was wondering how well Dependency-Track scales. According to the documentation, 16GB of RAM and 4 CPU cores are recommended for the API server. Does this number increase a lot when the API server is at high load (e.g. analyzing 100 projects simultaneously) or is it programmed in a way that it doesn't use more resources when the load increases?

Maybe a more concrete question: could Dependency-Track be used by an organization with about 500 projects? And have there been any performance tests to back up this claim?

Thanks in advance!

**Comment by stevespringett** [comment, marked as answer]
You'll want an instance that prioritizes CPU and RAM, not just has these things. Also of importance is the database instance that's chosen. DT can come to a crawl if the performance of the database server isn't up to par. 

Having 500+ projects is no problem. The biggest bottleneck currently are metric updates which will be improving in the next release and will likely be improved further the release after that. Portfolios with 1000+ projects are common.

**Reply by stevespringett** [reply]
Also, it should be noted that DT will attempt to use as much CPU and RAM as you give it, up to 90% of available RAM, and it will utilize as many cores as the machine has. It will not exceed the limits, however, it's important that the host is configured to allow such limits. For example, Docker Desktop defaults to only allowing containers to have 2GB RAM. If you give DT 16GB, it will crash.

**Reply by AbdelHajou** [reply]
Thanks for the quick response, good to know that it can handle large portfolios. Is the bottleneck in metric updates noticeable for the user?

**Reply by stevespringett** [reply]
Not noticeable to users per se, but if the metric updates take awhile it can slow down processing of portfolio and project metrics.
